File: 9-palindrome-number/9-palindrome-number.c

bool isPalindrome(int x){

    if (x<0 || (x % 10 == 0) && x!=0){
        return false;
    }
    int revertedNumber = 0;
    while(x>revertedNumber){
        revertedNumber = revertedNumber*10 + x % 10;
        x = x/10;
    }
    return ((revertedNumber == x ) || (revertedNumber / 10 == x));
}

File: 9-palindrome-number/9-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        String num = ""+x;
        for (int i = 0; i < num.length()/2;i++){
            if (num.charAt(i) !=num.charAt(num.length()-1-i)){
                return false;
            }
        }
        return true; 
    }
}